game-app/game-core/src/main/java/games/strategy/engine/framework/map/file/system/loader/ZippedMapsExtractor.java

```diff
@@ -88,22 +87,22 @@ public void unzipMapFiles() {
    * @param mapZip The map zip file to be extracted to the downloaded maps folder.
    * @return Returns extracted location (if successful, otherwise empty)
    */
-  public static Optional<File> unzipMap(final File mapZip) {
+  public static Optional<Path> unzipMap(final Path mapZip) {
     try {
       return unzipMapThrowing(mapZip);
     } catch (final IOException e) {
-      log.warn(
-          "Error extracting file: {}, {}", mapZip.getAbsolutePath() + ", " + e.getMessage(), e);
+      log.warn("Error extracting file: {}, {}", mapZip.toAbsolutePath() + ", " + e.getMessage(), e);
       return Optional.empty();
     }
   }
 
-  private static Optional<File> unzipMapThrowing(final File mapZip) throws IOException {
-    Preconditions.checkArgument(mapZip.isFile(), mapZip.getAbsolutePath());
-    Preconditions.checkArgument(mapZip.exists(), mapZip.getAbsolutePath());
-    Preconditions.checkArgument(mapZip.getName().endsWith(".zip"), mapZip.getAbsolutePath());
+  private static Optional<Path> unzipMapThrowing(final Path mapZip) throws IOException {
+    Preconditions.checkArgument(Files.isReadable(mapZip), mapZip.toAbsolutePath());
```

Comment:
Ditto on 'isReadable' != 'isFile'.

Unless there is a better API, this might need to be `!Files.isDirectory(mapZip))`
It could be worthwhile to wrap the File APIs in our own FileUtils so we that we will always have a consistent API regardless of implementation.

Reply:
This is a good thing to discuss actually.
There is `Files#isRegularFile`, which I hesitate to use because I ran into some issues with it a while ago.
For example `/dev/stdin` is not a regular file, even though it behaves like one and can be read just fine.
On the other hand if `mapZip` is a a file the current user isn't allowed to access `!Files.isDirectory(mapZip))` would obviously return true but fail somewhere down the line because of that.

So if you ask me checking if we can read the file is a better check for a precondition IMO, but I'm open to changing it

Reply:
I think this still needs to be resolved. The most likely error and reason for these as precondition checks is to guard against a programming error that we pass to this method a directory, instead of a zip file. The file name check seemingly will help a lot and the only way to have this method process a directory is if that directory has a '.zip' suffix (unlikely). I think there is value in having the check be explicit (and I will admit the original documentation within code was lacking), I think is very valuable.
In essence before we were checking:
- we are processing a file, not a directory
- the file exists
- the file is suffixed with .zip

The three conditions were meant to be a pretty good guarantee that we are in fact processing a zip file. One can actually go further and read the leading bits of a zip file to be really sure, though I did not do that here as it was arguably more complicated than worth it.

TL;DR, please add a simple check that we are processing a directory in addition to the other checks we have here. A check that we can 'read' the directory is a useful addition but not necessarily sufficient.

Reply:
@RoiEXLab if you would do this as a follow-up, would be appreciated. I'll go ahead and merge now considering this is relatively minor (would like to see it be done), but it'll be an easy to review and submit one-liner if so. Meanwhile any 'part 3' would be unblocked :+1: